src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -488,16 +488,13 @@ public TypeInfo(AssemblyInfo assembly, TypeDefinitionHandle typeHandle, TypeDefi
             this.type = type;
             methods = new List<MethodInfo>();
             Name = metadataReader.GetString(type.Name);
-            if (type.IsNested)
+            var declaringType = type;
+            while (declaringType.IsNested)
             {
-                var declaringType = metadataReader.GetTypeDefinition(type.GetDeclaringType());
-                Name = metadataReader.GetString(declaringType.Name) + "/" + Name;
-                Namespace = metadataReader.GetString(declaringType.Namespace);
-            }
-            else
-            {
-                Namespace = metadataReader.GetString(type.Namespace);
+                declaringType = metadataReader.GetTypeDefinition(declaringType.GetDeclaringType());
+                Name = metadataReader.GetString(declaringType.Name) + "." + Name;
```

Comment:
Why the switch from `"/"` to `"."` for nested types?

If the change is needed then, it will a need corresponding change in https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs#L504 else breakpoints will fail to resolve. And some tests will need to be updated.

Reply:
This is how VS shows a nested class in a console app. So we decided to follow the same idea. ExceptionThays2 is a class defined inside ExceptionThays.
![image](https://user-images.githubusercontent.com/4503299/143117141-9f0ca1fc-80d3-440d-9b5c-d8b06941acc7.png)